src/main/scala/org/broadinstitute/dsde/workbench/leonardo/model/LeonardoModel.scala

```diff
@@ -276,9 +274,64 @@ object LeonardoJsonSupport extends SprayJsonSupport with DefaultJsonProtocol {
 
   implicit val ClusterErrorFormat = jsonFormat3(ClusterError.apply)
 
-  implicit val ClusterFormat = jsonFormat22(Cluster.apply)
-
   implicit val DefaultLabelsFormat = jsonFormat6(DefaultLabels.apply)
 
   implicit val ClusterInitValuesFormat = jsonFormat20(ClusterInitValues.apply)
+
+  implicit object ClusterFormat extends RootJsonFormat[Cluster] {
+    override def read(json: JsValue): Cluster = {
+      json match {
+        case JsObject(fields: Map[String, JsValue]) =>
+          Cluster(
+            fields("id").convertTo[Long],
+            fields("clusterName").convertTo[ClusterName],
+            fields("googleProject").convertTo[GoogleProject],
+            fields("serviceAccountInfo").convertTo[ServiceAccountInfo],
+            DataprocInfo(fields("googleId").convertTo[Option[UUID]],
+                          fields("operationName").convertTo[Option[OperationName]],
+                          fields("stagingBucket").convertTo[Option[GcsBucketName]],
+                          fields("hostIp").convertTo[Option[IP]]),
+            AuditInfo(fields("creator").convertTo[WorkbenchEmail],
+                      fields("createdDate").convertTo[Instant],
+                      fields("destroyedDate").convertTo[Option[Instant]],
+                      fields("dateAccessed").convertTo[Instant]),
+            fields("machineConfig").convertTo[MachineConfig],
+            fields("clusterUrl").convertTo[URL],
+            fields("status").convertTo[ClusterStatus],
+            fields("labels").convertTo[LabelMap],
+            fields("jupyterExtensionUri").convertTo[Option[GcsPath]],
```

Comment:
I don't think this will work as intended -- if the "jupyterExtensionUri" key is not present in the Map, it will throw a `NoSuchElementException` instead of (correctly) deserializing to `None`. 

After some experimentation, I think the `fields` map should be accessed as follows:
```
fields.getOrElse("optionalKey", JsNull).convertTo[Option[A]]
fields.getOrElse("requiredKey", JsNull).convertTo[A]
```
I believe this will work correctly for all the cases:
- If the field is optional and absent, it will deserialize to None
- if the field is optional and present, it will deserialze to Some(a)
- if the field is required and absent, it will throw a `DeserializationException`
- if the field is required and present, it will deserialize to A